tree/ntuple/v7/src/RField.cxx

```diff
@@ -2368,6 +2394,50 @@ void ROOT::Experimental::RVariantField::CommitCluster()
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RSetField::RSetField(std::string_view fieldName, std::unique_ptr<Detail::RFieldBase> itemField)
+   : ROOT::Experimental::RProxiedCollectionField(fieldName, "std::set<" + itemField->GetType() + ">")
+{
+}
+
+std::unique_ptr<ROOT::Experimental::Detail::RFieldBase>
+ROOT::Experimental::RSetField::CloneImpl(std::string_view newName) const
+{
+   auto newItemField = fSubFields[0]->Clone(fSubFields[0]->GetName());
+   return std::make_unique<RSetField>(newName, std::move(newItemField));
+}
+
+void ROOT::Experimental::RSetField::ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value)
+{
+   if (!fProxy->GetCollectionClass()->HasDictionary()) {
+      throw RException(R__FAIL("RField: no dictionary loaded for collection type " + GetNormalizedType(fProxy->GetCollectionClass()->GetName()).first));
+   }
+
+   ClusterSize_t nItems;
+   RClusterIndex collectionStart;
+   fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nItems);
+
+   TVirtualCollectionProxy::TPushPop RAII(fProxy.get(), value->GetRawPtr());
+   fProxy->Clear();
+   void *item;
+   if (fProxy->GetType())
+      item = malloc(fItemSize);
+   else
+      item = fProxy->GetValueClass()->New();
+
+   for (unsigned i = 0; i < nItems; ++i) {
+      auto itemValue = fSubFields[0]->CaptureValue(item);
+      fSubFields[0]->Read(collectionStart + i, &itemValue);
+      fProxy->Insert(item, value->GetRawPtr(), 1);
```

Comment:
We may optimize this using bulk insertion (see one of my previous commits [here](https://github.com/root-project/root/commit/0e83aee79ad9423de5ffb70498526b95e1a0f63d)).  Still not sure whether it's worth it, given that for `std::set` we are not in high performance terrain anyways.

Reply:
I'd add a todo to future investigation.  I don' think we gain much for sets, where each insertion involves walking the tree and sorting.

Reply:
See https://github.com/root-project/root/pull/12948#discussion_r1254434423.  In principle we should be able to completely get rid of this function :slightly_smiling_face:.